src/bitbucket/BitbucketAPI.ts

```diff
@@ -30,67 +30,186 @@ export class BitbucketAPI {
       merge_strategy: 'merge_commit',
     };
 
-    Logger.info('Merging pull request', {
+    Logger.info('Attempting to merge pull request', {
       namespace: 'bitbucket:api:mergePullRequest',
       pullRequestId,
       landRequestId,
       targetBranch,
       landRequestStatus,
       postRequest: { endpoint, ...data },
     });
-    // This is just defining the function that we will retry
-    const attemptMerge = async () =>
-      axios.post(
-        endpoint,
-        JSON.stringify(data),
-        await bitbucketAuthenticator.getAuthConfig(
-          jwtTools.fromMethodAndPathAndBody('post', endpoint, data),
-          axiosPostConfig,
-        ),
-      );
-
-    const onFailedAttempt = (failure: AxiosError & pRetry.FailedAttemptError) => {
-      const { response, attemptNumber, attemptsLeft } = failure;
-      const { status, statusText, headers, data } = response || ({} as Record<string, undefined>);
-
-      // NOTE: Do **NOT** log the whole failure object here, it will have the axios config in it
-      // which contains auth credentials
-
-      Logger.error('Merge attempt failed', {
-        namespace: 'bitbucket:api:mergePullRequest:onFailedAttept',
+
+    // Polls result of merge task if the merge takes more than 28 seconds
+    // API returns 202 if this is required
+    const pollTaskResult = async (pollUrl: string): Promise<any> => {
```

Comment:
You can add simple unit test to your function with API calls using axios - check the bundle size add-on test

Reply:
I think worth investigating a bit of time in unit test & think about the edge cases - for example when we have concurrency set

Reply:
Yeah I'll look into writing a couple unit tests. I tried to do it with cypress but was having some difficulties with mocking everything and felt like it was going to become a time sink. Might try with jest because I'm more familiar with writing unit tests that way.

Reply:
There is a [cypress-testing-libraries](https://github.com/testing-library/cypress-testing-library) but not sure how good it is - can go with Jest - easy to setup and write